### ISSUE TITLE
AttributeError: 'tqdm' object has no attribute 'pos' when run using threads

### ISSUE THREAD

**Issue by gramster** [tqdm/tqdm#510]
python 3.6.4 from Anaconda distribution
OS X 10.13.3
tqdm 4.11.2

I have the following code:

```python
def populate(desc, events, ids):
    remaining = len(ids)
    for _ in tqdm(range(0, ((remaining + 999) // 1000)), desc=desc):
        batch = ids[:1000]
        ids = ids[1000:]
        result = get_batch(batch)
        events.insert_many(result)
```

ids is a list of strings, and events is a MongoDB collection; I'm doing some processing to get data for each ID and then populating the collection with the data and want a progress bar.

The above is called from this code:

```python
        if nthreads > 1:
            threads = []
            chunksize = (len(ids) + nthreads - 1) // nthreads
            for i in range(0, nthreads):
                t = threading.Thread(target=populate, args=(str(i), events, ids[(chunksize * i):(chunksize * (i + 1))]))
                threads.append(t)
                t.start()
        else:
            populate('-', events, ids)
```

It works fine if `nthreads` is 1, but if it is greater than one I get:

```
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/Users/gwheeler/anaconda/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/gwheeler/anaconda/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "backfill_edd.py", line 14, in populate
    for _ in tqdm(range(0, ((remaining + 999) // 1000)), desc=desc):
  File "/Users/gwheeler/anaconda/lib/python3.6/site-packages/tqdm/_tqdm.py", line 732, in __init__
    self.pos = self._get_free_pos(self) if position is None else position
  File "/Users/gwheeler/anaconda/lib/python3.6/site-packages/tqdm/_tqdm.py", line 391, in _get_free_pos
    return max(inst.pos for inst in cls._instances
  File "/Users/gwheeler/anaconda/lib/python3.6/site-packages/tqdm/_tqdm.py", line 392, in <genexpr>
    if inst is not instance) + 1
AttributeError: 'tqdm' object has no attribute 'pos'
```

**Comment by Kiv** [comment]
I also get this error.

Python 3.6.3 from Anaconda
Ubuntu 14.04
tqdm 4.19.4

**Comment by kittinan** [comment]
I also get this error when i use tqdm with ThreadPoolExecutor

```
AttributeError: 'tqdm' object has no attribute 'pos'
```

Python 3.6.3 :: Anaconda custom (64-bit)
tqdm: 4.19.4
Ubuntu: 14.04

**Comment by guifereis** [comment]
I get the same error, see also #404, I can confirm seems to happen with a very fast trange inside a tqdm call.

**Comment by casperdcl** [comment]
Just checking, do you still get this error with `tqdm>=4.19.7`?

**Comment by guifereis** [comment]
My bad, I was on 4.19.5. I just ran some quick tests and it appears 4.19.7
does indeed fix it. I will let you know if it pops up again with .7!

**Comment by guifereis** [comment]
Actually, let me take that back. I still see this on 4.9.17 -- 'tqdm'
object has no attribute 'pos'. I am using a threadpool and an imap, and
each of those threads in turn is also using a threadpool with an imap, so I
am spawning e.g. 4 threads which each keep a pool of another 4 = 16 threads
total. tqdm does a remarkable job of displaying the nested bars -
incredible really - except for when the error happens. I do suspect it is
synchronization because if I mess with the workload to make the threads
last longer it happens less.

**Comment by casperdcl** [comment]
closed as duplicate of #323 - will be fixed soon there

**Comment by casperdcl** [comment]
happy to reopen if this is still an issue for `tqdm>4.29.0` - I think it should be fixed by #573 (78227f92dbafc97ea08e19b442d6d33f63591bf4)
